### ISSUE TITLE
PR tab titles should be NavLinks

### ISSUE THREAD

**Issue by abhinavrastogi-harness** [harness/canary#1282]
All tabs should be using NavLinks internally. This is important for accessibility, usability and performance. It allows all tabs to be opened in a new tab, and also avoid maintaining this in React State.

<img width="1144" alt="Image" src="https://github.com/user-attachments/assets/48fe7db1-f922-4dd2-adfa-986134da227b" />

It seems today our main RepoSummary tabs are indeed NavLinks, but the PR tabs are not. We should fix this before launch.
